### PR TITLE
Fix #5485 Create means for verifying Fragment Arguments

### LINKED ISSUE #5485: [Feature Request]: Create means for verifying Fragment Arguments
### Is your feature request related to a problem? Please describe.

We use Espresso `Intents.Intended` to verify activity intent params, but we also need to test the fragment arguments and saved state bundles for correcteness, and to prevent bugs during development.

### Describe the solution you'd like

An example implementation was provided in [Fix part of #5485: Create means for verifying Fragment Arguments #5606](https://github.com/oppia/oppia-android/pull/5606). 

### Describe alternatives you've considered
Not testing fragement arguments, but since we are working long term on improving coverage, and also wanting to catch regressions, this is not an option.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by subhajitxyz**
## Explanation
Fixes #5485
Added test for 10 fragments arguments and saveInstanceState.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by subhajitxyz** [comment]
@adhiamboperes ,PTAL.

**Comment by subhajitxyz** [comment]
@adhiamboperes , PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileResetPinFragmentTest.kt [hunk 1]
```diff
@@ -1006,6 +1009,38 @@ class ProfileResetPinFragmentTest {
     }
   }
 
+  @Test
+  fun testFragment_fragmentLoaded_verifyCorrectArgumentsPassed() {
+    ActivityScenario.launch<ProfileResetPinActivity>(
+      ProfileResetPinActivity.createProfileResetPinActivity(
+        context = context,
+        profileId = 0,
+        isAdmin = true
```
Comment (adhiamboperes) [inline app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileResetPinFragmentTest.kt:1018]: Is there an expectation that the app behaviour will be different if `isAdmin = false`? If so, can we add another test for that too?

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @subhajitxyz! I just had one comment. PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileResetPinFragmentTest.kt (hunk 1 above)
Comment (subhajitxyz) [reply app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileResetPinFragmentTest.kt:1018]: I'll review it and let you know.

Comment (subhajitxyz) [reply app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileResetPinFragmentTest.kt:1018]: Yes, There is little change in app behavior.
If isAdmin = false, it asks for 3 digit pin.
If isAdmin = true, it asks for 5 digit pin. 

I tested the argument with isAdmin = false, it passed Expresso and the robolectric test.
Should I add this test?

Comment (adhiamboperes) [reply app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileResetPinFragmentTest.kt:1018]: Yes, you need to add the test.

Comment (subhajitxyz) [reply app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileResetPinFragmentTest.kt:1018]: Ok.

Comment (subhajitxyz) [reply app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileResetPinFragmentTest.kt:1018]: Is this test name appropriate for the test? testFragment_fragmentLoaded_whenIsAdminFalse_verifyCorrectArgumentsPassed

**Review by adhiamboperes — APPROVED** [review]
Thanks @subhajitxyz! This LGTM.

**Pull request merged**